File: JS/twiceold.js

//The function takes in 2 integers as parameters: the first is the father's age, the second is the son's age
//The function must return the number of years ago that the father was twice the age of the son's current age
// 43, 13 - 17
// Double the son's age and then subtract the dad's age from the son's doubled age. If the son's doubled age is greater than or equal to the dad's age, subtract the son's doubled age with the dad's age

function twiceAsOld(dadYearsOld, sonYearsOld) {

	if (sonYearsOld * 2 >= dadYearsOld) {
		return (sonYearsOld * 2) - dadYearsOld
	} else {
			return dadYearsOld - (sonYearsOld * 2)
	}
}

// function twiceAsOld2(dadYearsOld, sonYearsOld) {
// 	return Math.abs(dadYearsOld - (sonYearsOld * 2))
// }

console.log(twiceAsOld(43, 13))
console.log(twiceAsOld(55, 17))
console.log(twiceAsOld(32, 9))
console.log(twiceAsOld(75, 32))
console.log(twiceAsOld(36, 7))
console.log(twiceAsOld(55, 30))

File: py/twiceold.py
#The function takes in 2 integers as parameters: the first is the father's age, the second is the son's age
#The function must return the number of years ago that the father was twice the age of the son's current age
# 43, 13 - 17
# Double the son's age and then subtract the dad's age from the son's doubled age. If the son's doubled age is greater than or equal to the dad's age, subtract the son's doubled age with the dad's age

def twice_as_old(dad_years_old, son_years_old):
	if (son_years_old * 2) >= dad_years_old:
		return (son_years_old * 2) - dad_years_old
	elif dad_years_old > (son_years_old * 2):
		return dad_years_old - (son_years_old * 2)
	

print(twice_as_old(36, 11))
print(twice_as_old(29, 0))
print(twice_as_old(44, 20))
print(twice_as_old(57, 38))